base/strings/util.jl

```diff
@@ -458,13 +458,15 @@ function lpad(
     s::Union{AbstractChar,AbstractString},
     n::Integer,
     p::Union{AbstractChar,AbstractString}=' ',
-) :: String
+)
+    stringfn = if any(isa.((s, p), Union{TaggedString, TaggedChar, SubString{<:TaggedString}}))
+        taggedstring else string end
```

Comment:
This is a bit of a code-smell to me.

Could this be a dispatch on `string`?

Reply:
Could you elaborate on what you mean by "dispatch on `string`"?

Reply:
should `string(args...)` with `any(a isa Union{TaggedString, TaggedChar, SubString{<:TaggedString}}) for a in args)` just be, taggedstring? Instead of `taggedstring` being a new function.

The kinds of `isa` check you are doing seems to be something that we normally express in dispatch.

Reply:
I asked this during the triage call, the comment was that I shouldn't violate `string(xs...)` returning a `String`. That said `string(x)` will return any `AbstractString` not string :shrug:

Reply:
I'm happy to do whatever, just so long as I'm not told to undo it :stuck_out_tongue: